sync/src/transaction_queue.rs

```diff
@@ -0,0 +1,622 @@
+// Copyright 2015, 2016 Ethcore (UK) Ltd.
+// This file is part of Parity.
+
+// Parity is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Parity is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Parity.  If not, see <http://www.gnu.org/licenses/>.
+
+// TODO [todr] - own transactions should have higher priority
+
+//! Transaction Queue
+
+use std::vec::Vec;
+use std::cmp::{Ordering};
+use std::collections::{HashMap, BTreeSet};
+use util::uint::{Uint, U256};
+use util::hash::{Address};
+use util::table::*;
+use ethcore::transaction::*;
+
+
+#[derive(Clone, Debug)]
+struct VerifiedTransaction {
+	tx: SignedTransaction,
+	nonce_height: U256
```

Comment:
Consider not storing this value at all, but inserting into a sorted collection based on the current height. Although this could be difficult with rust.